### DISCUSSION TITLE
ESP32_CAM

### DISCUSSION THREAD

**Discussion by ghost** [luc-github/ESP3D#658] [F.A.Q] [answered]
There is an ESP32_CAM Espressif. How can I use Marlin with firmware and via webui to see a camera image as well?

**Comment by luc-github** [comment, marked as answer]
ESP32_cam is supported in ESP3D 3.0, you need to configure also need to configure one panel in webui  3.0 for camera https://github.com/luc-github/ESP3D-WEBUI/issues/123#issuecomment-718693756

**Reply by ghost** [reply]
How to do it? Do you have a description?

**Reply by luc-github** [reply]
There is no documentation for 3.0 currently because it is still in alpha and no bandwidth for do/update it, but basicaly you need to configure the camera in configuration.h, comments on parameter are pretty clear IMHO, on webUI once you have flashed ESP3D there is a new option that should appear in panel settings :` ESP32-CAM` so select it (the path `/snap` should be automaticaly added)  and add a refresh of 0.2, - (not less) and it should work

**Reply by ghost** [reply]
So I put it in the ESP3D V3.0 Visual Studio Code. I set it up and then upload it to the ESP32_CAM module. What should I do with WEBUI afterwards? It does not allow you to compile the VSC program.

**Reply by luc-github** [reply]
you need to use the existing one : https://github.com/luc-github/ESP3D-WEBUI/blob/3.0/dist/printer/production/index.html.gz
and go to web UI setting to add a camera panel as mentioned above, check the link I gave it a screenshot of UI

**Reply by ghost** [reply]
I understand. But I can't compile ESP3D version 3.0. Errors occur.

**Reply by luc-github** [reply]
please open issue and fill template to know what is happening : https://github.com/luc-github/ESP3D/issues/new/choose

**Comment by ghost** [comment]
All I have to do now is figure out how to connect to the BTT SKR 2 motherboard.

**Reply by luc-github** [reply]
https://github.com/bigtreetech/SKR-2/blob/master/Hardware/BIGTREETECH%20SKR%202-Pin.pdf

GND 3.3v RxTx

**Comment by atsju** [comment]
See [here](https://github.com/luc-github/ESP3D/issues/737#issuecomment-1141038053) for v3.0-alpha2

@luc-github Please tag this one as FAQ
